test/unittest.cpp

```diff
@@ -24,12 +28,30 @@ bool TestMemoryLeaks() {
 	return test_memory_leaks;
 }
 
+bool SummarizeFailures() {
+	return summarize_failures;
+}
+
+size_t GetSummaryCounter() {
+	return ++failures_summary_counter;
+}
+
+std::ostringstream &GetSummary() {
+	static std::ostringstream summary;
```

Comment:
I think we want to wrap this with a mutex:
```c++
struct FailureSummary {
public:
    void AddSummary(const string &failure) {
        std::lock_guard<std::mutex> guard(lock);
        summary << failure;
    }
public:
    std::ostringstream summary;
    std::mutex lock;
};

Reply:
Or, looking at how we use the `<<` operator a lot, we could create this instead:
```c++
void SafeAppend(std::function<void(std::ostringstream &)> callback) {
    std::lock_guard<std::mutex> guard(lock);
    callback(summary);
}
```

used like:
```c++
failure_summary.SafeAppend([](std::ostringstream &oss) {
    oss << termcolor::red << termcolor::bold << "Mismatch on row " << current_row + 1 << ", column "
		    << result.ColumnName(current_column) << "(index " << current_column + 1 << ")" << std::endl
		    << termcolor::reset; 
});
```